gateway/sensor.go

```diff
@@ -374,21 +377,26 @@ func convertToBytes(key interface{}) ([]byte, error) {
 	return res, nil
 }
 
+// Close closes the gateway.
 func (g *Gateway) Close(ctx context.Context) error {
 	if g.rstPin != "" && g.bookworm != nil {
-		gpio.ResetGPIO(g.rstPin, *g.bookworm)
+		err := gpio.ResetGPIO(g.rstPin, *g.bookworm)
+		if err != nil {
+			g.logger.Error("error reseting gateway")
+		}
 	}
 
 	if g.workers != nil {
 		g.workers.Stop()
 	}
 	errCode := C.stopGateway()
```

Comment:
[q]Would printing the error code be useful for debugging

Reply:
no its always 1

Reply:
I have a ticket to better handle and surface the errors from the C code though so may readdress this there